torch/nn/init.py

```diff
@@ -0,0 +1,240 @@
+        xavier_uniform(tensor.data, gain=gain)
+        return tensor
+    else:
+        fan_in, fan_out = _calculate_fan_in_and_fan_out(tensor)
+        std = gain * np.sqrt(2.0 / (fan_in + fan_out))
+        a = np.sqrt(3.0) * std
+        return tensor.uniform_(-a, a)
+
+
+def xavier_normal(tensor, gain=1):
+    """Fills the input Tensor or Variable with values according to the method described in "Understanding the difficulty of training
+       deep feedforward neural networks" - Glorot, X. and Bengio, Y., using a normal distribution.
+
+       The resulting tensor will have values sampled from normal distribution with mean=0 and
+       std = gain * sqrt(2/(fan_in + fan_out))
+
+    Args:
+        tensor: a n-dimension torch.Tensor
+        gain: an optional scaling factor to be applied
+
+    Examples:
+        >>> w = torch.Tensor(3, 5)
+        >>> nninit.xavier_normal(w, gain=np.sqrt(2.0))
+    """
+    if isinstance(tensor, Variable):
+        xavier_normal(tensor.data, gain=gain)
+        return tensor
+    else:
+        fan_in, fan_out = _calculate_fan_in_and_fan_out(tensor)
+        std = gain * np.sqrt(2.0 / (fan_in + fan_out))
+        return tensor.normal_(0, std)
+
+
+def kaiming_uniform(tensor, gain=1):
+    """Fills the input Tensor or Variable with values according to the method described in "Delving deep into rectifiers: Surpassing
+       human-level performance on ImageNet classification" - He, K. et al using a uniform distribution.
+
+       The resulting tensor will have values sampled from U(-a, a) where a = gain * sqrt(1/(fan_in))
+
+    Args:
+        tensor: a n-dimension torch.Tensor
+        gain: an optional scaling factor to be applied
+
+    Examples:
+        >>> w = torch.Tensor(3, 5)
+        >>> nninit.kaiming_uniform(w, gain=np.sqrt(2.0))
+    """
+
+    if isinstance(tensor, Variable):
+        kaiming_uniform(tensor.data, gain=gain)
+        return tensor
+    else:
+        fan_in, _ = _calculate_fan_in_and_fan_out(tensor)
+        std = gain * np.sqrt(1.0 / fan_in)
+        a = np.sqrt(3.0) * std
+        return tensor.uniform_(-a, a)
+
+
+def kaiming_normal(tensor, gain=1):
+    """Fills the input Tensor or Variable with values according to the method described in "Delving deep into rectifiers:
+       Surpassing human-level performance on ImageNet classification" - He, K. et al using a normal distribution.
+
+       The resulting tensor will have values sampled from normal distribution with mean=0 and
+       std = gain * sqrt(1/(fan_in))
+
+    Args:
+        tensor: a n-dimension torch.Tensor
+        gain: an optional scaling factor to be applied
+
+    Examples:
+        >>> w = torch.Tensor(3, 5)
+        >>> nninit.kaiming_normal(w, gain=np.sqrt(2.0))
+    """
+    if isinstance(tensor, Variable):
+        kaiming_normal(tensor.data, gain=gain)
+        return tensor
+    else:
+        fan_in, _ = _calculate_fan_in_and_fan_out(tensor)
+        std = gain * np.sqrt(1.0 / fan_in)
```

Comment:
Yeah, it should be `sqrt(2.0 / n)` (the 2 instead of a 1 is the big takeaway from the paper). What's the point of gain? Isn't the point of the initialization that it preserves magnitudes? Setting a gain breaks that.